modyn/storage/internal/file_wrapper/csv_file_wrapper.py

```diff
@@ -0,0 +1,198 @@
+import csv
+from typing import Iterator, Optional
+
+from modyn.storage.internal.file_wrapper.abstract_file_wrapper import AbstractFileWrapper
+from modyn.storage.internal.file_wrapper.file_wrapper_type import FileWrapperType
+from modyn.storage.internal.filesystem_wrapper.abstract_filesystem_wrapper import AbstractFileSystemWrapper
+
+
+class CsvFileWrapper(AbstractFileWrapper):
+    def __init__(self, file_path: str, file_wrapper_config: dict, filesystem_wrapper: AbstractFileSystemWrapper):
+        super().__init__(file_path, file_wrapper_config, filesystem_wrapper)
+
+        self.file_wrapper_type = FileWrapperType.CsvFileWrapper
+
+        if "separator" in file_wrapper_config:
+            self.separator = file_wrapper_config["separator"]
+        else:
+            self.separator = ";"
+
+        if "label_index" not in file_wrapper_config:
+            raise ValueError(
+                "Please specify the index of the column that contains the label. "
+                "Use None if no column contains the label"
+            )
+        self.label_index = file_wrapper_config["label_index"]
+
+        # the first line might contain the header, which is useless and must not be returned.
+        if "ignore_first_line" in file_wrapper_config:
+            self.ignore_first_line = file_wrapper_config["ignore_first_line"]
+        else:
+            self.ignore_first_line = False
+
+        if "encoding" in file_wrapper_config:
+            self.encoding = file_wrapper_config["encoding"]
+        else:
+            self.encoding = "utf-8"
+
+        # check that the file is actually a CSV
+        self._validate_file_extension()
+
+        self._validate_file_content()
```

Comment:
To answer your question: It would be great to make this configurable, whether these checks should be performed, and default to `True`.

Reply:
Ok, now it's a parameter `validate_file_content`